# 基礎測驗

說明: 首先這個檔案是僅供檢視的，所以你必須在自己的雲端硬碟中另存副本，再開始進行你自己的回答。回答過程中可以使用ChatGPT, 最後需要附上ChatGPT的使用連結。(在ChatGPT右上角的Share Chat，把連結複製下來貼在最下面。為什麼要最下面呢？因為這個Share Chat必須是在你完成了所有問題的詢問之後再給出來，才會有完整的問答)

每一題有不同的分數，而每一題當中，程式碼都要有註解。請以文字框，把你的想法寫出來。評分將包含程式能否正確完成，有無註解，以及該回答的文字框中答案的完整性。

最後請將這個檔案下載成『你的學號_Quiz3.ipynb』檔案，並上傳到moodle繳交完成測驗。

以下這個文字框是要填寫的：

** 學號: AN4126521**

** 姓名: 江婕瀅**

** 系級: 不分 116**

# 第一題，做一個學生的物件：
請用Python做一個class，它可以在初始的時候放進去學生的學號、姓名、以及它的自我介紹

In [5]:
class Student:
  def __init__(self, id: str, name: str, introduction: str):
    print(f"My name is {name}, id is {id}, {introduction}")


student_A = Student("AN4126521", "Jesse", "I love program.")
student_B = Student("an4126521", "Jessie", "I from CCEP.")
student_C = Student("An4126521", "Jessi", "My age is 21.")


My name is Jesse, id is AN4126521, I love program.
My name is Jessie, id is an4126521, I from CCEP.
My name is Jessi, id is An4126521, My age is 21.


# 第二題，數字和的平方和
1. 請寫出一個物件An，這個物件An中可以放一個value，有一個next方法可以產生它value的每一個數字平方的總和
2. 請把An的next算出來的數字再放入value中
3. 請試著多做幾次，你覺得An會有什麼規律？


In [23]:
class An:
  def __init__(self, value: str):
    self.value = value

  def next(self):
    temp = 0
    for i in self.value:
      temp += int(i)**2

    self.value = str(temp)
    print(self.value)


an = An(input("請輸入一個數字："))

is_run = True
while is_run:

  an.next()

  is_again = input("是否再算一次？（Y/N）")
  if (is_again == "N" or is_again == "n"):
    is_run = False

請輸入一個數字：2
4
是否再算一次？（Y/N）Y
16
是否再算一次？（Y/N）Y
37
是否再算一次？（Y/N）Y
58
是否再算一次？（Y/N）Y
89
是否再算一次？（Y/N）Y
145
是否再算一次？（Y/N）Y
42
是否再算一次？（Y/N）Y
20
是否再算一次？（Y/N）Y
4
是否再算一次？（Y/N）N


任何大於一的數最後都會切回到 4 > 16 > 37 > 58 > 89 > 145 > 42 > 20 這個循環

# 第三題：請以字典方式寫出今天import過的模組
1. 請至少寫出三個你自己今天import過的模組，以及它對應的功用。並且把它放入一個名為My_Module_Learning的字典dict中。

In [24]:
My_Module_Learning = {0: "沒有 inport module"}

print(My_Module_Learning)

{0: '沒有 inport module'}


# 第四題：自行設計
請自行設計一個物件，你得說明這個物件你的設計想法，然後跟你的生活有什麼關連性，這個物件至少必須包含了兩個屬性及一個方法。

說明:

In [40]:
class Phone:
  def __init__(self, phone_number: str):
    self.phone_number = phone_number
    self.contact = {}

  def add_contact(self, person_name: str, person_number: str):
    print(f"add {person_name}")
    self.contact[person_name] = person_number

  def show_contact(self):
    print(f"Show all contact：{self.contact}")


phone = Phone(phone_number="0903655705")
print(f"My phone number is {phone.phone_number}\n")

phone.add_contact(person_name="J", person_number="0911111111")
phone.show_contact()

phone.add_contact(person_name="E", person_number="0999999999")
phone.show_contact()

phone.add_contact(person_name="S", person_number="0988888888")
phone.show_contact()

My phone number is 0903655705

add J
Show all contact：{'J': '0911111111'}
add E
Show all contact：{'J': '0911111111', 'E': '0999999999'}
add S
Show all contact：{'J': '0911111111', 'E': '0999999999', 'S': '0988888888'}
